### PR TITLE
patrons: edit the user personal data in a modal

### PR DESCRIPTION
**Pull request by jma**
* Opens an editor in a modal dialog to edit the user personal
  information, from the patron editor of the professional interface.
* Adapts patron detailed and profile view for new fields.
* Makes the patron barcode repeatable.
* Adds a custom user ID editor wrapper.

Co-Authored-by: Johnny Mariéthoz <Johnny.Mariethoz@rero.ch>

## Why are you opening this PR?

- Which task/US does it implement?
- Which issue does it fix?

## Dependencies

My PR depends on `rero-ils#<xx>`'s PR(s):

* https://github.com/rero/ng-core/pull/354
* https://github.com/rero/rero-ils/pull/1695

## How to test?

- What command should I have to run to test your PR?
- What should I test through the UI?

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

File: projects/admin/src/app/record/brief-view/patrons-brief-view.component.ts [hunk 1]
```diff
@@ -27,7 +27,7 @@ import { ResultItem } from '@rero/ng-core';
         {{ record.metadata.last_name }}, {{ record.metadata.first_name }}
       </a>
       <small class="ml-3" *ngIf="record.metadata.patron">
-        <a id="{{record.metadata.patron.barcode + '-loans'}}" [routerLink]="['/circulation', 'patron', record.metadata.patron.barcode]">
+        <a id="{{record.metadata.patron.barcode[0] + '-loans'}}" [routerLink]="['/circulation', 'patron', record.metadata.patron.barcode[0]]">
```
Comment (Garfield-fr) [inline projects/admin/src/app/record/brief-view/patrons-brief-view.component.ts:30]: cosmetic: space before `{` and after `}`

File: projects/admin/src/app/record/custom-editor/user-id-editor/user-id-editor.component.ts [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+  /** current query to import a user */
+  searchText: string = null;
+
+  /** current User id in the invenio db */
+  userID: string = null;
+
+  /** remotely loaded User id in the invenio db */
+  loadedUserID: string = null;
+
+  /** JOSONSchema */
```
Comment (Garfield-fr) [inline projects/admin/src/app/record/custom-editor/user-id-editor/user-id-editor.component.ts:44]: Typo: `JSONSchema `

File: projects/admin/src/app/record/custom-editor/user-id-editor/user-id-editor.component.ts [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+    }
+    this._recordService.getRecords('users', query).pipe(
+      map((res: any) => {
+        if (res.hits.hits.length === 0) {
+          this._toastService.warning(
+            this._translateService.instant('User not found.')
+          );
+          return null;
+        }
+
```
Comment (Garfield-fr) [inline projects/admin/src/app/record/custom-editor/user-id-editor/user-id-editor.component.ts:139]: cosmetic: remove line

File: projects/admin/src/app/record/editor/wrappers/user-id/user-id.component.html [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  GNU Affero General Public License for more details.
+
+  You should have received a copy of the GNU Affero General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+-->
+<ng-container *ngIf="user$ | async as user; else create">
+  <strong>
+    {{user.metadata.last_name}}, {{ user.metadata.first_name }} <span *ngIf="user.metadata.city">({{ user.metadata.city}})</span>
```
Comment (Garfield-fr) [inline projects/admin/src/app/record/editor/wrappers/user-id/user-id.component.html:19]: cosmetic: `{{ user.metadata.last_name }}` and `{{ user.metadata.city }}`

**Review by Garfield-fr — APPROVED** [review]

**Review by iGormilhit — APPROVED** [review]

**Pull request merged**